Make priority hints tests more stable

rwlbuis opened the pull request:
#### ed46651c79801e40b8e2d1fa5fd53d9dad7af48f
<pre>
Make priority hints tests more stable
<a href="https://bugs.webkit.org/show_bug.cgi?id=254885">https://bugs.webkit.org/show_bug.cgi?id=254885</a>

Reviewed by Fujii Hironori.

Make two priority hints tests stable. To fix fetch-api-priority.html
in addition we need getResourcePriority to fallback to the memory
cache, since gc could clear the document resources in CachedResourceLoader
at any time.

* LayoutTests/http/tests/priority-hints/fetch-api-priority.html:
* LayoutTests/http/tests/priority-hints/link-header-fetch-priority.py:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::isLoadingFromMemoryCache):
(WebCore::Internals::resourceFromMemoryCache):
(WebCore::Internals::getResourcePriority):
* Source/WebCore/testing/Internals.h:

Canonical link: <a href="https://commits.webkit.org/262883@main">https://commits.webkit.org/262883@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a31b83053adfd795a19f6a4bb24c30c3cc7b940a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/2842 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/2911 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/3001 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/4254 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/3276 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/2809 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/2992 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/2950 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/2497 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/2870 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/3267 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/2534 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/4046 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/762 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/2517 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/2393 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/2522 "Passed tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/2568 "5 failures") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/3788 "run-api-tests-without-change (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/2923 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/2333 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/2572 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/2520 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/718 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/2563 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/2743 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on LayoutTests/http/tests/priority-hints/fetch-api-priority.html:
```diff
@@ -51,8 +52,9 @@
                     test.request_info.url : test.request_info;
-      const response = await fetch(test.request_info, test.request_init);
-      checkResourcePriority(url, test.expected_priority);
+      fetch(test.request_info, test.request_init).then(function(response) {
```
You probably want to either do `await fetch` or `return fetch`

fujii replied:
Can you revert this part?

rwlbuis committed:
Make priority hints tests more stable
https://bugs.webkit.org/show_bug.cgi?id=254885

Reviewed by Fujii Hironori.

Make two priority hints tests stable. To fix fetch-api-priority.html
in addition we need getResourcePriority to fallback to the memory
cache, since gc could clear the document resources in CachedResourceLoader
at any time.

* LayoutTests/http/tests/priority-hints/fetch-api-priority.html:
* LayoutTests/http/tests/priority-hints/link-header-fetch-priority.py:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::isLoadingFromMemoryCache):
(WebCore::Internals::resourceFromMemoryCache):
(WebCore::Internals::getResourcePriority):
* Source/WebCore/testing/Internals.h:

Canonical link: https://commits.webkit.org/262883@main
